CMakeLists.txt

```diff
@@ -705,40 +794,56 @@ set(ConfigPackageLocation ${CMAKE_INSTALL_LIBDIR}/cmake/libtorrent-rasterbar)
 string(REGEX REPLACE "([^;]+)" "find_dependency(\\1)" _find_dependency_calls "${_package_dependencies}")
 string(REPLACE ";" "\n" _find_dependency_calls "${_find_dependency_calls}")
 
+if(CMAKE_VERSION VERSION_LESS "3.11.0")
+	set(_compatibility ExactVersion)
+else()
+	set(_compatibility SameMinorVersion)
+endif()
+
 write_basic_package_version_file(
-	"${CMAKE_CURRENT_BINARY_DIR}/libtorrent-rasterbar/libtorrent-rasterbarConfigVersion.cmake"
+	"${CMAKE_CURRENT_BINARY_DIR}/libtorrent-rasterbar/libtorrent-rasterbar-config-version.cmake"
 	VERSION ${libtorrent_VERSION}
-	COMPATIBILITY ExactVersion
+	COMPATIBILITY ${_compatibility}
 )
 
-export(EXPORT torrent-rasterbarTargets
-	FILE "${CMAKE_CURRENT_BINARY_DIR}/libtorrent-rasterbar/libtorrent-rasterbarTargets.cmake"
+export(EXPORT libtorrent-rasterbar-targets
+	FILE "${CMAKE_CURRENT_BINARY_DIR}/libtorrent-rasterbar/libtorrent-rasterbar-targets.cmake"
 	NAMESPACE Libtorrent::
```

Comment:
@arvidn, @ssiloti: there is a bit of inconvenience here: we export target "torrent-rasterbar" into namespace "Libtorrent". At the same time, package name of the library is "libtorrent-rasterbar". Thus the cmake find module is named "FindLibtorrentRasterbar.cmake". However, these export commands create imported target in "Libtorrent" namespace. For consistency "rasterbar" should be moved into the namespace name, and the full name of the exported target would become "LibtorrentRasterbar::torrent-rasterbar" then. 

Let's look at the library file names now. Consider Linux only. The CMake build creates output library named "libtorrent-rasterbar.so" At the same time, b2 build seems to create output "libtorrent.so" (at least I found no mention for "rasterbar" in the Jamfile), autotools build creates "libtorrent-rasterbar.la" (and .so, naturally). 

As you see, there are various naming schemes and I would suggest to unify them. My question is: what are the proper names for the package and output library?

Reply:
Both the package and library should be named "libtorrent-rasterbar".

Reply:
Since cmake default to camelCased names, "FindLibtorrentRasterbar.cmake" and target named "LibtorrentRasterbar::torrent-rasterbar", i.e. exporting to namespace "LibtorrentRasterbar" should be OK?

Reply:
Seems reasonable

Reply:
OK, commited these changes.